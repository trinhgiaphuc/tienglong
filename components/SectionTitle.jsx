import Link from 'next/link';

const SectionTitle = ({ title, animation1, animation2, href = '/' }) => (
  <Link href={href} passHref>
    <div className="my-border flex-center group cursor-pointer ">
      {animation1}
      <h2 className="title-responsive font-medium px-4 text-center py-20 uppercase">
        {title}
      </h2>
      {animation2}
    </div>
  </Link>
);

export default SectionTitle;
