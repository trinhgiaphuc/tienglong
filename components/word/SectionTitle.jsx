import Link from 'next/link';

const SectionTitle = ({ children, href = '/' }) => {
  return (
    <Link href={href} passHref>
      <div className="my-border flex-center group py-10  cursor-pointer">
        {children}
      </div>
    </Link>
  );
};

export default SectionTitle;
