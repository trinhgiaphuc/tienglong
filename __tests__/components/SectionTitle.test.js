import SectionTitle from "@components/word/SectionTitle";
import { render, screen } from "@testing-library/react";

describe('<SectionTitle', () => {
  test('render the section title', () => {
    const { container, debug } = render(<SectionTitle>Fucked Up</SectionTitle>);

    const div = container.querySelector('div');
    debug(div);
    expect(div).toHaveClass('my-border');
    expect(div).toHaveTextContent('Fucked Up');
  });

  test('renders and matches the snapshot', () => {
    const { container, debug } = render(<SectionTitle>Fucked Up</SectionTitle>);

    expect(container).toMatchSnapshot();

  });

});

