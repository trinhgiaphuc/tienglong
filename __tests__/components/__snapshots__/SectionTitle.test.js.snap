// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionTitle renders and matches the snapshot 1`] = `
<div>
  <div
    class="my-border flex-center text-center text-5xl font-bold font-ole group py-10"
  >
    Fucked Up
  </div>
</div>
`;
